jsontableschema/infer.py

```diff
@@ -84,3 +85,86 @@ def infer(headers, values, row_limit=None, explicit=False, primary_key=None):
         schema['fields'][index].update(**rv)
 
     return schema
+
+
+# Internal
+
+class _TypeGuesser(object):
+
+    """Guess the type for a value.
+
+    Returns:
+        * A tuple  of ('type', 'format')
+
+    """
+
+    def __init__(self, type_options=None):
+        self._types = _get_available_types()
+        self.type_options = type_options or {}
+
+    def cast(self, value):
+        for _type in reversed(self._types):
+            result = _type(self.type_options.get(_type.name, {})).test(value)
+            if result:
+                # TODO: do format guessing
```

Comment:
@roll please open an issue for this. I believe this comment is probably from my original code? In any event, I've noticed in OpenSpending over the last months how useful it would be to guessing full type + format, and not just type, so this is good to capture in an issue (cc @akariv FYI)